File: redsheet-portal/src/app/database-models/v-back-ground-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vBackGroundRecord {
    constructor(vBackGroundRecord?: vBackGroundRecord )  {
        let properties = [ "focus", "negotiationTypeId", "negotiationTypeName", "opponentName", "projectBackgroundId", "projectModuleId", "reason", "valueObjectiveId", "valueObjectiveName",];
        for (let property of properties)
            if (vBackGroundRecord && vBackGroundRecord[property])
                this[property] = vBackGroundRecord[property];
    }
 
    @required()
    @maxLength(1000)
	focus : string =   undefined;
 
	negotiationTypeId : number =   0 ;
 
    @required()
    @maxLength(200)
	negotiationTypeName : string =   undefined;
 
    @required()
    @maxLength(1000)
	opponentName : string =   undefined;
 
    @range(1,2147483647)
	projectBackgroundId : number =   undefined;
 
    @range(1,2147483647)
	projectModuleId : number =   undefined;
 
    @required()
    @maxLength(1000)
	reason : string =   undefined;
 
    @range(1,2147483647)
	valueObjectiveId : number =   undefined;
 
    @required()
    @maxLength(200)
	valueObjectiveName : string =   undefined;


}

File: RedSheet.DbEntities/Models/vBackGroundRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vBackGroundRecords", Schema = "dbo")]
	[Serializable]
	public partial class vBackGroundRecord  
	{
 
	[Required] 
	[MaxLength(1000)]
	public string Focus { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int NegotiationTypeId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string NegotiationTypeName { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string OpponentName { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectBackgroundId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectModuleId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Reason { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ValueObjectiveId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string ValueObjectiveName { get; set; } 

	public vBackGroundRecord()
	{
    }

    }
}

